src/controllers/tasks.ts

```diff
@@ -0,0 +1,20 @@
+import { Response } from 'express';
+import { CustomError } from '../helper';
+import { getTasksByUserId } from '../database/query/tasks';
+import { TokenRequest } from '../interfaces';
+
+const getTasks = (req: TokenRequest, res: Response): void => {
+  const userId = req.userData?.id;
+  getTasksByUserId(+userId!)
+    .then((tasks) => {
+      res.status(200).json({
+        message: 'Tasks retrieved successfully',
+        data: tasks,
+      });
+    })
+    .catch(() => {
+      throw new CustomError(500, 'Server Error');
```

Comment:
throw error from the catch to where? use `next` here to send the error to the serverError middleware
"we use throw error in the `then` section `ya 3asal`"